src/geometry/lane_geometry.cc

```diff
@@ -68,11 +98,24 @@ maliput::math::RollPitchYaw LaneGeometry::Orientation(double p) const {
 }
 
 maliput::math::RollPitchYaw LaneGeometry::Orientation(const maliput::math::Vector3& prh) const {
-  MALIPUT_THROW_MESSAGE("LaneGeometry::Orientation(prh) is not implemented yet.");
+  // TODO: Take into account r and h displacement.
+  return Orientation(prh.x());
 }
 
 maliput::math::Vector3 LaneGeometry::WInverse(const maliput::math::Vector3& xyz) const {
-  MALIPUT_THROW_MESSAGE("LaneGeometry::WInverse is not implemented yet.");
+  const utility::ClosestPointResult closest_centerline_point = utility::GetClosestPoint(centerline_, xyz);
+  const double p = closest_centerline_point.p;
+
+  const maliput::math::Vector3 d_xyz = xyz - closest_centerline_point.point;
+
+  const maliput::math::Vector3 s_hat = utility::GetTangentAtP(centerline_, closest_centerline_point.p);
+  const maliput::math::Vector3 r_hat =
+      Orientation(closest_centerline_point.p).ToMatrix() * maliput::math::Vector3::UnitY();
+  const maliput::math::Vector3 h_hat = s_hat.cross(r_hat);
```

Comment:
Because we know that the orientation remains the same at any r and h for a given p. You can get the columns of the rotation matrix and get each direction vector. In other words, I suggest you pick the Orientation(closest_centerline_point.p) and get the second column to build r_hat and the third column to build h_hat.

Reply:
You are right, I can directly use the orientation matrix to get r and h.
Done.